### ISSUE TITLE
execution failed suddenly with no apparent reason

### ISSUE THREAD

**Issue by masihurmaruf** [spotify/bucket-runner#6]
I have a test suit which run a bunch of tests based on a db query. Query returns an array and i am using that array to run the the tests. But the tests are taking a good amount of time. So tried to use `bucket-runner` but unfortunately the execution stops suddenly with no apparent reason(at least i cannot think of any reason). 

I am using `bucket-runner --stream-output --partition-size 5 tests/critique/critique-test.js -- ./node_modules/.bin/mocha` this command to run my tests. And only thing i see after the execution crashed is this `/usr/local/lib/node_modules/bucket-runner/bin/cli.js:59
    if (err) throw err;
SIGTERM`

any help will be much appreciated. Thanks.

**Comment by kirbysayshi** [comment]
I'm not sure what the problem is. But one thing you can do to diagnose is to run the command with debug logging on: `DEBUG=* bucket-runner --stream-output --partition-size 5 tests/critique/critique-test.js -- ./node_modules/.bin/mocha`.

Your command though is only running using one input file though. bucket-runner is fine with this, but you're also running with `partition-size` of 5. Partition size tells bucket-runner to group files, but there is only one file... so I'm not sure how it would behave in this situation. Maybe try setting that to `--partition-size 1` as well? bucket-runner isn't very useful with just one file!

**Comment by masihurmaruf** [comment]
Thanks for the reply. In my situation to decrease the execution time, can you please suggest anything to me?

**Comment by kirbysayshi** [comment]
If you give bucket-runner multiple files and use `--partition-size=1`, then it will spawn one process per file, and in parallel. While it depends on your tests and what exactly they do, this is generally faster than running all the files sequentially. It's unclear from your original question though: do you have multiple files?

**Comment by masihurmaruf** [comment]
No. I have a test file with one test method. I am using the db query to reuse that single test.

**Comment by kirbysayshi** [comment]
How are the results of the DB query being passed to the test file? Again, bucket-runner cannot speed up a single file. It's only when there are multiple files, since it can split them over parallel processes.
